src/js/containers/search/filters/KeywordContainer.jsx

```diff
@@ -0,0 +1,70 @@
+/**
+  * KeywordContainer.jsx
+  * Created by Emily Gullo 11/30/16
+  **/
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import * as searchFilterActions from 'redux/actions/search/searchFilterActions';
+
+import Keyword from 'components/search/filters/keyword/Keyword';
+
+const propTypes = {
+    updateTextSearchInput: React.PropTypes.func,
+    clearFilterType: React.PropTypes.func,
+    keyword: React.PropTypes.string
+};
+
+export class KeywordContainer extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            newKeyword: '',
+            keyword: ''
+        };
+
+        this.submitText = this.submitText.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(e) {
+        this.setState({
+            newKeyword: e.target.value
+        });
+    }
+
+    submitText() {
+        // remove any previous keyword applied in redux
+        if (this.props.keyword !== null) {
```

Comment:
This first `if` statement is unnecessary. Redux, as a state machine, will simply overwrite the old keyword with the new keyword, so there is no reason to clear the filter type.